src/Build.UnitTests/Evaluation/ProjectCollection_Tests.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using Microsoft.Build.Evaluation;
+using Shouldly;
+using Xunit;
+
+namespace Microsoft.Build.UnitTests.Evaluation
+{
+    public class ProjectCollection_Tests
+    {
+        [Fact]
+        public void ProjectRootElementCache_IsDeterminedByEnvironmentVariable()
+        {
+            var collectionWithDefaultCache = new ProjectCollection();
+            collectionWithDefaultCache.ProjectRootElementCache.ShouldBeOfType<ProjectRootElementCache>();
+
+            const string envKey = "MsBuildUseSimpleProjectRootElementCacheConcurrency";
+            string originalEnvVar = Environment.GetEnvironmentVariable(envKey);
+
+            try
+            {
+                Environment.SetEnvironmentVariable(envKey, "true");
+
+                var collectionWithSimpleCache = new ProjectCollection();
+                collectionWithSimpleCache.ProjectRootElementCache.ShouldBeOfType<SimpleProjectRootElementCache>();
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(envKey, originalEnvVar);
```

Comment:
nit: You can use TestEnvironment to smoothly define environment variables that get automatically reverted when the TestEnvironment gets disposed.